client/src/components/Dialogs/index.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import AddRoleDialog from './AddRoleDialog';
+
+const DIALOG_COMPONENTS = {
+  'ADD_ROLE': AddRoleDialog,
+};
+
+const DialogRoot = ({ dialog }) => {
+  if (!dialog.type) {
+    return null;
+  }
+
+  const SpecificDialog = DIALOG_COMPONENTS[dialog.type];
```

Comment:
This component is the only that gets rendered in the app.js. It will import all other dialogs and determine which one to render depending on what is the the dialog store. Each dialog component will need to have their own unique name which will be used in the action payload to open the dialog.